core/domain/story_services.py

```diff
@@ -623,6 +623,13 @@ def update_story(
         new_exploration_context_models)
     exp_models.ExplorationContextModel.put_multi(new_exploration_context_models)
 
+    # Generate new TopicSummary after a Story has been edited to
+    # make sure the TopicSummaryTile displays the correct number
+    # of chapters.
+    topic_id = new_story.corresponding_topic_id
+    from core.domain import topic_services
+    topic_services.generate_topic_summary(topic_id)
```

Comment:
@aks681 I've moved the code we're discussing to here (bottom of `update_story`). Note the inline import to avoid circular import.

Reply:
Oh, I don't think we should be doing inline imports anywhere, since it then gets difficult to keep track of the files imported in a service (I remember @seanlip advising against this once, maybe he can weigh in as well here). So, I'd still suggest doing what I initially suggested.

Reply:
I agree with avoiding inline imports, yup.